src/diagram/models/variable.ts

```diff
@@ -261,7 +261,7 @@ export const Variable = types.model("Variable", {
           return {unit: unitString};
         }
       } else {
-        if (self.inputs[0]?.computedUnit || self.inputs[1]?.computedUnit) {
+        if (nodeInputs[0]?.computedUnit || nodeInputs[1]?.computedUnit) {
```

Comment:
This should be changed to a loop that goes over all of the inputs. If any of the inputs have a unit then we show the "units cancel" message:

```suggestion
        if (nodeInputs.some(input => input?.computedUnit) {
```